pkg/reconciler/common/stages.go

```diff
@@ -96,3 +100,20 @@ func DeleteObsoleteResources(ctx context.Context, instance v1alpha1.KComponent,
 		return installed.Filter(mf.NoCRDs, mf.Not(mf.In(*manifest))).Delete()
 	}
 }
+
+// SetErrorConditionIfIndeterminate marks the instance's Conditions as not Ready
+// if no existing error has been signaled. If Ready is already False, it does
+// not change the existing error reporting. This exists to ensure that errors
+// are reported to the user even if the underlying functions do not explicitly
+// set conditions on the instance.
+func SetErrorConditionIfIndeterminate(err error, instance v1alpha1.KComponent) {
+	if !instance.GetStatus().GetCondition(apis.ConditionReady).IsFalse() {
+		instance.GetStatus().SetConditions(apis.Conditions{
+			apis.Condition{
+				Type:    apis.ConditionReady,
+				Status:  v1.ConditionFalse,
+				Reason:  "StageError",
+				Message: "Internal error: " + err.Error(),
+			}})
```

Comment:
Overwriting all the conditions with just a single one here makes me nervous. Is there no risk of losing any condition set by a previous stage?